docs/architecture/CLI-TEST-ARCHITECTURE.md

```
@@ -0,0 +1,502 @@
+# CLI-Based Test Architecture
+
+**Design Document for Migrating from Programmatic to CLI-Based Testing**
```

markdownlint-cli2 (0.18.1)

Emphasis used instead of a heading

(MD036, no-emphasis-as-heading)